### DISCUSSION TITLE
Trait bound `MethodRouter<B> Error

### DISCUSSION THREAD

**Discussion by SamuelBonilla** [tokio-rs/axum#835] [Q&A] [answered]
Hi, I'm trying to use the Router struct inside another struct but I'm getting this error on this line: 

``` rust
self.router = self.router.route(_path, get(handler));
```

the trait bound 'MethodRouter<B>: Service<hyper::Request<hyper::Body>>' is not satisfied
the trait 'Service<hyper::Request<hyper::Body>>' is not implemented for `MethodRouter<B>

this is the code:

``` rust
struct MyStruc {
    router: Router<Body>,
}

impl MyStruc  {

    pub fn new() -> Self {
        Self {
            router: Router::new(),
        }
    }

    pub fn myFunc<H, T, B>(&self, _path: &path::Path<&str>, handler: H)
    where
        H: Handler<T, B>,
        B: Send + 'static,
        T: 'static,
    
    {
    
        self.router = self.router.route(_path, get(handler));
    }   
}
```

what I'm doing wrong?

**Comment by davidpdrsn** [comment, marked as answer]
You're `Router` requires `Body` but the handler takes a generic `B`. You have to either use `Handler<T, Body>` or `Router<B>`. I'd recommend the former.

**Reply by SamuelBonilla** [reply]
thanks :)
